### ISSUE TITLE
include doesn't work when a value is provided several times

### ISSUE THREAD

**Issue by sorawee** [brownplt/pyret-lang#1132]
In `foo.arr`

```
provide {
  f: f,
  g: f,
} end

f = _ + 1
```

In the definition window:

```
include foo
import foo as F
```

In the REPL:

```
> f
<function:f>
> g
The identifier `g` is unbound
> F.f
<function:f>
> F.g
<function:f>
```

**Comment by sorawee** [comment]
Sorry, I think the issue is about providing name different from the value name rather than providing several times.

**Comment by blerner** [comment]
To be clear: the fact that the last two lines both print `f` as the function name is expected behavior. The fact that the second line doesn't work is a bug. The fact that `g: f` has two distinct names is a red herring, and not relevant here: we've routinely provided values at different names than they were defined with.

**Comment by sorawee** [comment]
Yes, the fact the the second line doesn't work is a bug, and it's because it's written as `g: f`. If it were to be written as `g: g`, the problem will not occur.

A real example that actually happen:

In the definition window:

```
include file
```

In the REPL:

```
> file-to-string
<function:file-to-string>
> real-path
unbound id
```

Despite:

```
provide {
  ...
  file-to-string: file-to-string,
  real-path: F.real-path
} end
```

in https://github.com/brownplt/pyret-lang/blob/master/src/arr/trove/file.arr

**Comment by blerner** [comment]
It seems like the include mechanism is behaving differently from the import mechanism. What happens if we include something that provides {x: 5}?  I.e. something that isn't even a name?

This bug just bit someone here trying to split their solution file into a teacher file and a student starter file, so I can confirm it's definitely happening :-(

**Comment by jpolitz** [comment]
Oof. Yup, this is badness in how names are calculated for provides.

This is related to https://github.com/brownplt/pyret-lang/issues/1009 and https://github.com/brownplt/pyret-lang/issues/1015 and https://github.com/brownplt/pyret-lang/wiki/TM005---More-structured-module-system,-to-facilitate-data,-renaming,-bringing-identifiers-into-local-scope

Basically, the left-hand-sides in provides using object literals are way too “stupid,” and non-`*` provides need attention.
